src/lock.rs

```diff
@@ -39,10 +39,19 @@ pub fn write_lock<P: AsRef<Path>>(
     lock.write(&lock_file)
 }
 
-fn hash_files<'a>(files: impl Iterator<Item = &'a PathBuf>) -> Fallible<BTreeMap<PathBuf, String>> {
+fn hash_files<'a, P: 'a>(files: impl Iterator<Item = &'a P>) -> Fallible<BTreeMap<String, String>>
+where
+    P: AsRef<Path>,
+{
     files
         .map(|p| match hash_file(p) {
-            Ok(hash) => Ok((p.clone(), hash)),
+            Ok(hash) => Ok((
+                p.as_ref()
+                    .to_str()
+                    .unwrap_or("non-uft8-path")
```

Comment:
Won't that fail when you read the lock file? Why not raise an error here?

Reply:
Yes, but the paths could also come from collecting during processing (used in `write_lock`), not only from the lock file. But maybe we should fail here?